### ISSUE TITLE
rework RTL for 64-bit compatibility.

### ISSUE THREAD

**Issue by LADSoft** [LADSoft/OrangeC#497]
rework the RTL for 64-bit compatibility.   This includes replacing the windows header files if possible.

**Comment by chuggafan** [comment]
Our best bet for this one is most likely, as I've suggested before, pulling from MINGW64, however I cam now recognize some of the issues with this as much of it is either: 

1. public domain or

2. LGPL

The latter is a problem however if we're trying to rebuild the MSVC stuff, the only other option is relying on a system install from MSVC or rebuilding the headers from windows, the latter of which is.... honestly something I can do, but would rather not spend the 2+ years on digging through the various MSVC docs again to get a type to have the exact structure layout it's supposed to have.

**Comment by LADSoft** [comment]
yeah replacing the windows headers is a priority here.    But we also need to go through and make sure pointers aren't being aliased to ints, and size_t is used where appropriate.
